src/os/MemStore.h

```diff
@@ -16,39 +16,58 @@
 #ifndef CEPH_MEMSTORE_H
 #define CEPH_MEMSTORE_H
 
-#include "include/assert.h"
+#include <mutex>
+#include <boost/intrusive_ptr.hpp>
+
 #include "include/unordered_map.h"
 #include "include/memory.h"
+#include "include/Spinlock.h"
 #include "common/Finisher.h"
+#include "common/RefCountedObj.h"
 #include "common/RWLock.h"
 #include "ObjectStore.h"
+#include "PageSet.h"
+#include "include/assert.h"
 
 class MemStore : public ObjectStore {
+private:
+  CephContext *const cct;
+
 public:
-  struct Object {
-    bufferlist data;
+  struct Object : public RefCountedObject {
+    std::mutex xattr_mutex;
+    std::mutex omap_mutex;
```

Comment:
Thanks for the review. If you take a look at MemStore::_clone() and it's use of std::lock() to safely take the locks of multiple objects, that's something that the Mutex interface doesn't provide.